examples/cloud-run/tgi-deployment/README.md

```diff
@@ -0,0 +1,348 @@
+
+```bash
+SERVICE_URL=$(gcloud run services describe $SERVICE_NAME --region $LOCATION --format 'value(status.url)')
+```
+
+Then you can send requests to the deployed service on Cloud Run, using the `SERVICE_URL` and any Google Cloud Credentials with Cloud Run Invoke access. For setting up the credentials there are multiple approaches, some of those are listed below:
+
+- Using the default identity token from the Google Cloud SDK:
+
+  - Via `gcloud` as:
+
+  ```bash
+  gcloud auth print-identity-token
+  ```
+
+  - Via Python as:
+
+  ```python
+  import google.auth
+  from google.auth.transport.requests import Request as GoogleAuthRequest
+
+  auth_req = GoogleAuthRequest()
+  creds, _ = google.auth.default()
+  creds.refresh(auth_req)
+
+  id_token = creds.id_token
+  ```
+
+- Using a Service Account with Cloud Run Invoke access, which can either be done with any of the following approaches:
+  - Create a Service Account before the Cloud Run Service was created, and then set the `--service-account` flag to the Service Account email when creating the Cloud Run Service. And use an Access Token for that Service Account only using `gcloud auth print-access-token --impersonate-service-account=SERVICE_ACCOUNT_EMAIL`.
+  - Create a Service Account after the Cloud Run Service was created, and then update the Cloud Run Service to use the Service Account. And use an Access Token for that Service Account only using `gcloud auth print-access-token --impersonate-service-account=SERVICE_ACCOUNT_EMAIL`.
+
+The recommended approach is to use a Service Account (SA), as the access can be controlled better and the permissions are more granular; as the Cloud Run Service was not created using a SA, which is another nice option, you need to now create the SA, gran it the necessary permissions, update the Cloud Run Service to use the SA, and then generate an access token to set as the authentication token within the requests, that can be revoked later once you are done using it.
+
+0. Set the `SERVICE_ACCOUNT_NAME` environment variable for convenience:
+
+```bash
+export SERVICE_ACCOUNT_NAME=text-generation-inference-invoker
+```
+
+1. Create the Service Account:
+
+```bash
+gcloud iam service-accounts create $SERVICE_ACCOUNT_NAME
+```
+
+2. Grant the Service Account the Cloud Run Invoker role:
+
+```bash
+gcloud run services add-iam-policy-binding $SERVICE_NAME \
+    --member="serviceAccount:$SERVICE_ACCOUNT_NAME@$PROJECT_ID.iam.gserviceaccount.com" \
+    --role="roles/run.invoker" \
+    --region=$LOCATION
+```
+
+3. Update the Cloud Run service to use the Service Account:
+
+```bash
+gcloud run services update $SERVICE_NAME \
+    --service-account $SERVICE_ACCOUNT_NAME@$PROJECT_ID.iam.gserviceaccount.com \
+    --region=$REGION
+```
+
+4. Generate the Access Token for the Service Account:
+
+```bash
+export ACCESS_TOKEN=$(gcloud auth print-access-token --impersonate-service-account=$SERVICE_ACCOUNT_NAME@$PROJECT_ID.iam.gserviceaccount.com)
```

Comment:
Where do you expect customers to run this gcloud command to get the access otken?

How to get token really depends on the use cases, I would suggest following the guide here:

https://cloud.google.com/run/docs/authenticating/overview

Reply:
Great I'll add a reference to it, while mentioning these two alternatives here! Ideally this is intended to run from a terminal on an user's instance or on a Google Cloud instance with the required access. But more focused on the user's perspective than pure production; I'll add the reference and some other comments anyway.